source/dubregistry/internal/workqueue.d

```diff
@@ -49,13 +50,27 @@ final class PackageWorkQueue {
 		}
 	}
 
+	void putFront(string pack_name)
+	{
+		synchronized (m_mutex) {
+			m_queue.putFront(pack_name);
```

Comment:
In the first commit (https://github.com/dlang/dub-registry/pull/291/commits/28ba1327e40afe318c1baf53760dd60444af68b9), this used to do search for the package in the buffer and swap it to the front if it can be found. As Martin already used a `static if (is(typeof({ m_queue.putFront(pack_name); })))` blocked, I assumed that this was a preparation for the new Vibe.d version and just kept this block.
(This might lead to a subsequent update if a manual update was requested.)

Reply:
Maybe we should at least check for `m_queue[].count(pack_name) < 2`, so that spamming the update queue infinitely is not possible as easily.

Reply:
why would you want to update it multiple times? just the canFind as below would make sense here imo

Reply:
Just `canFind` would break the initial motivation for this, namely to be able to trigger an immediate update, regardless of the state of the general update queue. Ideally, if scheduled, it should be removed from its current spot and be put at the front instead, but having it be updated a second time during the normal update cycle still sounds completely reasonable.

Reply:
oh yeah, hm I think moving it would make more sense than updating it twice, all these git providers have a rate limit after all

Reply:
Although the rate-limit for GitHub only applies for non-cached responses, so we are generally fine there. We could also stretch out the regular update interval to an hour to reduce the general stress a bit.

Reply:
> Maybe we should at least check for m_queue[].count(pack_name) < 2, so that spamming the update queue infinitely is not possible as easily.

Went with `if (!m_queue[0 .. min(10, $)].canFind(pack_name))` - seems a bit more robust, but it's still "equally naive".
A proper protection would probably check that a package isn't updated/triggered more than X times within one hour ...